tests/test_foundation/test_network_types.py

```diff
@@ -17,6 +17,72 @@ def test_GenTLV_value_unpack(self):
         self.assertEqual(tlv.value.value, tlv_unpacked.value.value)
 
 
+class TestEthernet(unittest.TestCase):
+    """Test Ethernet frames."""
+
+    def test_Ethernet_pack(self):
```

Comment:
You can define your variables as class attributes, so you will only need to define them once.
you can have `untagged_ethernet`, `tagged_ethernet`, `untagged_expected`, `tagged_expected`.

Also, I think that's a nice idea to do the Fail test on this case.